dev-support/create-release/release-build.sh

```diff
@@ -20,114 +20,180 @@
 trap cleanup EXIT
 
 # Source in utils.
-SELF=$(cd $(dirname $0) && pwd)
+SELF="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+# shellcheck source=SCRIPTDIR/release-util.sh
 . "$SELF/release-util.sh"
 
 # Print usage and exit.
 function exit_with_usage {
-  cat << EOF
-Usage: release-build.sh <build|publish-snapshot|publish-release>
-Creates build deliverables from a tag/commit.
-Arguments:
- build             Create binary packages and commit to dist.apache.org/repos/dist/dev/hbase/
- publish-snapshot  Publish snapshot release to Apache snapshots
- publish-release   Publish a release to Apache release repo
-
-All other inputs are environment variables:
- GIT_REF - Release tag or commit to build from
- PACKAGE_VERSION - Release identifier in top level package directory (e.g. 2.1.2RC1)
- VERSION - (optional) Version of project being built (e.g. 2.1.2)
- ASF_USERNAME - Username of ASF committer account
- ASF_PASSWORD - Password of ASF committer account
- GPG_KEY - GPG key used to sign release artifacts
- GPG_PASSPHRASE - Passphrase for GPG key
- PROJECT - The project to build. No default.
-
-Set REPO environment to full path to repo to use
-to avoid re-downloading dependencies on each run.
+  cat <<'EOF'
+Usage: release-build.sh <tag|publish-dist|publish-snapshot|publish-release>
+Creates release deliverables from a tag or commit.
+Argument: one of 'tag', 'publish-dist', 'publish-snapshot', or 'publish-release'
+  tag               Prepares for release on specified git branch: Set release version,
+                    update CHANGES and RELEASENOTES, create release tag,
+                    increment version for ongoing dev, and publish to Apache git repo.
+  publish-dist      Build and publish distribution packages (tarballs) to Apache dist repo
+  publish-snapshot  Build and publish maven artifacts snapshot release to Apache snapshots repo
+  publish-release   Build and publish maven artifacts release to Apache release repo, and
+                    construct vote email from template
+
+All other inputs are environment variables.  Please use do-release-docker.sh or
+do-release.sh to set up the needed environment variables.  This script, release-build.sh,
+is not intended to be called stand-alone, and such use is untested.  The env variables used are:
```

Comment:
Do you think we need an entrance script or a rename of the one of these scripts as main.sh or some such. As is, operator has to consult the README. I suppose this ok given how specialized this all is.

Reply:
I was letting it go due to how specialized it is, but you're right; there's no reason not to make it nice and clear.  Will try to improve this; see comment immediately above.

Reply:
Sweet

Reply:
Getting late, will tackle this in the morning.